plugin/src/main/java/org/wildfly/plugin/provision/ApplicationImageMojo.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2022, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.wildfly.plugin.provision;
+
+import static java.lang.String.format;
+import static java.lang.String.join;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.Arrays;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+
+/**
+ * Build (and push) an application image containing the provisioned server and the deployment.
+ *
+ * The {@code}image{@code} goal extends the {@code}package{@code} goal, building and pushing the image occurs after the server
+ * is provisioned and the deployment deployed in it.
+ *
+ * The {@code}image{@code} goal relies on a Docker binary to execute all image commands (build, login, push).
+ *
+ * @since 4.0
+ */
+@Mojo(name = "image", requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME, defaultPhase = LifecyclePhase.PACKAGE)
```

Comment:
I am wandering if we could have an option to disable actual build of the image but still generate the Dockerfile. A kind of dry-run, allowing to build docker image outside of maven or use a maven plugin.

Reply:
That's a good idea. The Dockerfile is generated in the `target`. I could generate it in any case when the `image` goal is run and check for `image.build` after its generation.

Reply:
done, @jfdenise thanks for the suggestion.